File: app/src/main/java/com/boardgamegeek/sorter/PlayersSorterFactory.java
package com.boardgamegeek.sorter;

import android.content.Context;
import android.support.annotation.NonNull;

public class PlayersSorterFactory {
	public static final int TYPE_NAME = 1;
	public static final int TYPE_QUANTITY = 2;
	public static final int TYPE_DEFAULT = TYPE_NAME;

	@NonNull
	public static PlayersSorter create(@NonNull Context context, int type) {
		switch (type) {
			case TYPE_QUANTITY:
				return new PlayersQuantitySorter(context);
			case TYPE_NAME:
			default:
				return new PlayersNameSorter(context);
		}
	}
}

File: app/src/main/java/com/boardgamegeek/sorter/PlayersSorterFactory.kt
package com.boardgamegeek.sorter

import android.content.Context

class PlayersSorterFactory {

    companion object {
        const val TYPE_NAME = 1
        const val TYPE_QUANTITY = 2
        const val TYPE_WINS = 3
        const val TYPE_DEFAULT = TYPE_NAME

        @JvmStatic
        fun create(context: Context, type: Int): PlayersSorter {
            return when (type) {
                TYPE_QUANTITY -> PlayersQuantitySorter(context)
                TYPE_WINS -> PlayersWinSorter(context)
                TYPE_NAME -> PlayersNameSorter(context)
                else -> PlayersNameSorter(context)
            }
        }
    }
}
